docs/architecture.md

```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
 
 The agent communicates with the OSConfig Management Platform over the Management Platform Interface (MPI) IPC REST API.
 
 <img src="assets/agent.png" alt="OSConfig Agent" width=50%/>
 
-The agent is completely decoupled from the platform and the modules. A new module can be installed, and necessary PnP interface(s) published as part of the OSConfig Model and that will be enough to make OSConfig use the respective module, without the need to recompile the agent or the platform.
+The agent is completely decoupled from the platform and the modules. A new module can be installed, and, optionally, PnP interface(s) published as part of the OSConfig Model and that will be enough to make OSConfig use the respective module, without the need to recompile the agent or the platform.
+
+## 3.2. RC/DC Watcher
+
+The RC/DC Watcher monitors the Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. The Watcher also makes MpiGetReported calls to the Management Platform and writes the reported configuration to the Reported Configuration (RC) file. 
 
-## 3.2. AIS Client
+The RC/DC files are written in MIM JSON format.
 
-The agent  uses the PnP C SDK's HTTP helper APIs to make GET and POST requests to the [Azure Identity Services (AIS)](https://azure.github.io/iot-identity-service/) (identity, key, certificates) to build the device/module identity to (re)connect to the IoT Hub with. 
+To protect against unauthorized access, the RC/DC files are restricted to root user read and write access only.
 
-## 3.3. IoT Hub Client
+The  RC/DC files by default reside under /etc/osconfig/ as `/etc/osconfig/osconfig_desired.json` (the DC file) and `/etc/osconfig/osconfig_reported.json` (the RC file).
 
-The agent's  IoT Hub Client parses the Desired Twin payload received from the IoT Hub to separate individual PnP property values. The agent does not attempt to parse the property values. 
+## 3.3. GitOps Watcher
 
-## 3.4. Watcher
+The RC/DC Watcher clones a Git repository, monitors the GitOps Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. 
 
-The Watcher monitors the Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. The Watcher also makes MpiGetReported calls to the Management Platform and writes the reported configuration to the Reported Configuration (RC) file. 
+The GitOps DC file by default is named `osconfig_desired.json` and is located in the root of the repository at `/etc/osconfig/gitops/osconfig_desired.json`.
 
-The pair of DC and RC files are written in MIM JSON format. The RC/DC files act as local Device Twins. 
+The GitOps DC file is written in MIM JSON format.

```

Comment:
nit: consider adding link to [mim.schema.json](https://github.com/Azure/azure-osconfig/blob/ahbenmes/moduletest_distro_onboard/src/modules/schema/mim.schema.json) here

Reply:
Please review last commit. Refresh.